src/util/windows.jl

```diff
@@ -0,0 +1,66 @@
+"""
+An iterator which moves across a container, as a sliding window.
+"""
+immutable EachWindowIterator{T <: AbstractArray}
```

Comment:
Ah yeah I forgot to ask about this - what do people think of making the Sequence abstract type inherit from AbstractArray? I think it makes some sense, a sequence can be thought of as an ordered array of nucleotides, it has getindex, setindex, length, start, next, done and most of the other methods set out in the documentation on the [Array Interface](http://docs.julialang.org/en/release-0.4/manual/interfaces/)